EB-699: Run worker container as non-root

brinkdp opened the pull request:
This PR is a companion-piece to the k3d work done in the private deployment repo. It started out as a collection of small tweaks to simplify using `divbase-cli` with the k3d cluster, but the lasting achievement here is that the worker dockerfile is now configured to run as non-root which had previously been a challenge on MacOS.

In short, the previous permissions issues that has plagued several MacOS docker use-cases were solved by switching from Docker Desktop to Rancher Desktop. Apparently Rancher Desktop handles local permissions in a better way. In the worker dockerfile, GID and UID is now configured and the user is given access to `/app` so that bcftools can write its temp and results files. 

Another thing discovered was that the fastapi image does not become healthy with bcrypt v5.0.0 which was released this week. It's pinned to v4.3.0 for now to avoid the bug.

The updates to `scripts/local_dev_setup.py` triggered with the `--k3d` flag were useful when playing around with nodePorts in the k8s manifests. But now that we have switched to port-forwarding, the updates to the script and the idea of using `.env.local-k3d` might not even be necessary since we can just as well use the same ports as we would use for the Docker Compose stack. If you want we could rollback those commits.

RMCrean reviewed (approved):
Great job, as said on the other PR, it is working nicely for me now! :) 

Glad you managed to solve the non-root issue too! That will save lots of headaches in the future I suspect.

> The updates to scripts/local_dev_setup.py triggered with the --k3d flag were useful when playing around with nodePorts in the k8s manifests. But now that we have switched to port-forwarding, the updates to the script and the idea of using .env.local-k3d might not even be necessary since we can just as well use the same ports as we would use for the Docker Compose stack. If you want we could rollback those commits.

Feel free to decide, as you prefer. I think it is unlikely we will use the local k8s build much after the initial setup so that speaks towards a simpler scripts/local_dev_setup.py, which does not explicitly cater to k8s. That said if you prefer having the prefix on the bucket names/ability to clearly distinguish them I'm not opposed to it.

brinkdp commented:
> I think it is unlikely we will use the local k8s build much after the initial setup so that speaks towards a simpler scripts/local_dev_setup.py, which does not explicitly cater to k8s. 

This is a good point! Also felt nice to clean up the code. I reverted the script and also dropped the `.env.local-k3d` since the port-forwarding is compatible with `DIVBASE_ENV=local`.

As a sanity-check, I have ascertained that the set of test commands I used for the k3d cluster work after the revert:
```
python scripts/local_dev_setup.py

DIVBASE_ENV=local divbase-cli dimensions update --project local-project-1

DIVBASE_ENV=local divbase-cli query bcftools-pipe --tsv-filter "Area:Northern Portugal" --command "view -s SAMPLES; view -r 21:15000000-25000000" --metadata-tsv-name sample_metadata.tsv --project local-project-1

DIVBASE_ENV=local divbase-cli query task-status

DIVBASE_ENV=local divbase-cli files list --project local-project-1
```

Will go ahead and merge now.
